library/src/main/java/com/pokegoapi/api/settings/Settings.java

```diff
@@ -123,6 +126,30 @@ public void updateSettings(DownloadSettingsResponse response) {
 		if (response.getSettings().hasGpsSettings()) {
 			gpsSettings.update(response.getSettings().getGpsSettings());
 		}
+
+		// check if new api has been forced by Niantic
+		if(!response.getSettings().getMinimumClientVersion().isEmpty()) {
+			int[] currentVer = api.getHashProvider().getCurrentAPIVersion();
+			String[] minimumVer = response.getSettings().getMinimumClientVersion().split("\\.");
+			int[] splitMinimumVer = new int[currentVer.length];
```

Comment:
`minimumVer` should be `splitMinimumVer`. What is currently `spltMinimumVer` should not use the length of `currentVer`, but the length of what is currently `minimumVer`. You should probably also handle the case of `currentVer` and `minimumVer` having different lengths.